src/TSP/BranchAndBound.java

```diff
@@ -0,0 +1,186 @@
+package TSP;
+import java.util.ArrayList;
+
+public class BranchAndBound implements Algoritme{
+
+    public ArrayList<Pakket> calculatePath(ArrayList<Pakket> pakketjesOriginal) {
+        ArrayList<Pakket> pakketjes = (ArrayList<Pakket>) pakketjesOriginal.clone();
+        int pathLength = pakketjes.size();
+        ArrayList<RouteWrapper> routes = new ArrayList<RouteWrapper>();
+
+        for (int i = 1; i <= pathLength; i++) {
+            int[] initialRoute = new int[pathLength];
+            initialRoute[0] = i;
+
+            routes.add(new RouteWrapper(initialRoute, pakketjes));
+        }
+        int[] testpath = new int[pathLength];
+
+        for (int i = 0; i < pathLength; i++) {
+            testpath[i] = i + 1;
+        }
+        routes.add(new RouteWrapper(testpath, pakketjes));
+
+
+
+        double upperbound=Double.POSITIVE_INFINITY;
+        while (moreRoutesToSearch(routes,upperbound) == true) {
+            RouteWrapper routeToCheck = determineNextNode(routes,upperbound);
+            if(routeToCheck!=null) {
+                routes.addAll(lookForNextNodes(routeToCheck, pakketjes));
+
+//                System.out.println(Arrays.toString(routeToCheck.getLocations()));
+            }
+            routes.remove(routeToCheck);
+            upperbound = findNewUpperbound(routes);
+//            System.out.println(upperbound);
+//            if(FindShortestCompleteRoute(routes)!=null) {
+//                System.out.println(FindShortestCompleteRoute(routes).getLength());
```

Comment:
dit soort dingen uiteindelijk helemaal weghalen uit de code of in een debugger functie ofzo